msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -110,11 +164,11 @@ public override bool Execute ()
 			if (Resources.Length == 0)
 				return true;
 
-			EnvironmentVariables = new string[] {
-				"CODESIGN_ALLOCATE=" + CodesignAllocate
-			};
+			Parallel.ForEach (Resources, new ParallelOptions { MaxDegreeOfParallelism = 4 }, (item) => {
```

Comment:
I'll need to test again (to be 100%) sure but recall only testing the project (which I checked) on device (not on a sim). However you're right that *all* simulator builds (even release) should be done without a timestamp.